distributed/tests/py3_test_asyncio.py

```diff
@@ -0,0 +1,167 @@
+# flake8: noqa
+
+import pytest
+
+asyncio = pytest.importorskip('asyncio')
+aiohttp = pytest.importorskip('aiohttp')  # pytest-aiohttp
+
+import sys
+from operator import add
+from dask import delayed
+from toolz import isdistinct
+from distributed.deploy import LocalCluster
+from distributed.utils_test import gen_cluster, inc, div, slowinc, slowadd, slowdec, randominc
+
+from distributed.asyncio import AioClient, AioFuture
+from distributed import Client, Worker, Scheduler
+
+from tornado.platform.asyncio import to_asyncio_future, BaseAsyncIOLoop
+from tornado.ioloop import IOLoop
+
+
+@delayed
+def decrement(x):
+    return x - 1
+
+
+async def test_asyncio_submit(loop):
+    async with AioClient(loop=loop, processes=False) as c:
+        x = c.submit(inc, 10)
+        assert not x.done()
+
+        assert isinstance(x, AioFuture)
+        assert x.client is c
+
+        result = await x.result()
+        assert result == 11
+        assert x.done()
+
+        y = c.submit(inc, 20)
+        z = c.submit(add, x, y)
+
+        result = await z.result()
+        assert result == 11 + 21
+
+
+async def test_asyncio_map(loop):
+    async with AioClient(loop=loop, processes=False) as c:
+        L1 = c.map(inc, range(5))
+        assert len(L1) == 5
+        assert isdistinct(x.key for x in L1)
+        assert all(isinstance(x, AioFuture) for x in L1)
+
+        result = await L1[0].result()
+        assert result == inc(0)
+
+        L2 = c.map(inc, L1)
+
+        result = await L2[1].result()
+        assert result == inc(inc(1))
+
+        total = c.submit(sum, L2)
+        result = await total.result()
+        assert result == sum(map(inc, map(inc, range(5))))
+
+        L3 = c.map(add, L1, L2)
+        result = await L3[1].result()
+        assert result == inc(1) + inc(inc(1))
+
+        L4 = c.map(add, range(3), range(4))
+        results = await c.gather(L4)
+        if sys.version_info[0] >= 3:
```

Comment:
I was just copying a couple of test cases from test_client. This still needs more thorough testing. Right now I'm trying to get green on travis :)